ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -913,7 +913,35 @@ public func zip<S: SequenceType, T, Error where S.Generator.Element == SignalPro
 	return .empty
 }
 
+extension SignalProducer where T: SignalProducerType, E == T.E {
+	/// Flattens the inner producers sent upon `producer` (into a single producer of
+	/// values), according to the semantics of the given strategy.
+	///
+	/// If `producer` or an active inner producer emits an error, the returned
+	/// producer will forward that error immediately.
+	///
+	/// `Interrupted` events on inner producers will be treated like `Completed`
+	/// events on inner producers.
+	@warn_unused_result(message="Did you forget to call `start` on the producer?")
+	public func flatten(strategy: FlattenStrategy) -> SignalProducer<T.T, E> {
```

Comment:
Hm, yes I was over-engineering. Actual change was as simple as replace ```SignalProducer``` to ```SignalProducerType``` and retaining the rest for flatten as is. it compiles, it works as it pass the tests.

There are many other things that could be moved to ```SignalProducerType``` and ```SignalType``` in the similar manner, but I would prefer to keep it in a separate PR in other case it will grow to big.